### PR TITLE
Split selfservice flag

### PR DESCRIPTION
**Pull request by babamishra9**
[UAA] Apply split "self-service" flag to existing UAA zone configs

### CODE REVIEW

File: model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java [hunk 1]
```diff
@@ -98,25 +98,35 @@ public Logout setWhitelist(List<String> whitelist) {
     }
 
     public static class SelfService {
-        private boolean selfServiceResetPasswordEnabled = true;
-        private boolean selfServiceCreateAccountEnabled = true;
+        private boolean selfServiceLinksEnabled = true;
```
Comment (Yogeshkumarb57) [inline model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java:101]: why we added selfServiceLinksEnabled back and converted type of ther flags to String?

File: model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java [hunk 2]
```diff
@@ -98,25 +98,35 @@ public Logout setWhitelist(List<String> whitelist) {
...
+        private String selfServiceResetPasswordEnabled = null;
```
Comment (Yogeshkumarb57) [inline model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java:102]: why we converted type of selfServiceResetPasswordEnabled and  selfServiceCreateAccountEnabled to String?

File: model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java (hunk 1 above)
Comment (babamishra9) [reply model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java:101]: Compatibility for existing zones, selfServiceLinksEnabled field has been added back.

File: model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java (hunk 2 above)
Comment (babamishra9) [reply model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java:102]: To accommodate all the backward compatibility use cases and selfservice split flags for new zones creation cases, to have the three different value for selfServiceResetPasswordEnabled and selfServiceCreateAccountEnabled as null, "true", "false".

**Comment by irinaepshteyn** [comment]
@babamishra9 , the overall feeling I get is that solution could be simplified.  I can't put my finger on a better solution yet, but looks like there are two entry points where conversion from one single flag to split-flags could happen:
- when zone configuration is written to the database: https://github.com/GESoftware-CF/uaa/blob/rc_75.18.1/server/src/main/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioning.java#L94-L98
- when zone configuration is read from the database: https://github.com/GESoftware-CF/uaa/blob/rc_75.18.1/server/src/main/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioning.java#L159-L166

Once old single-flag config is mapped to the new split-flag config, we could start referring to the two flags everywhere else, i.e. in LoginInfoEndpoint, just like you did in the first commit.

File: model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java (hunk 2 above)
Comment (Yogeshkumarb57) [reply model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java:102]: You can use wrapper type Boolean for all three values null, true and false.

Comment (babamishra9) [reply model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java]: Done. Thanks!

**Comment by akarollil** [comment]
Superseded by https://github.com/GESoftware-CF/uaa/pull/441

**Pull request closed without merging**